src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/service/ActivityMappingService.kt

```diff
@@ -82,12 +82,64 @@ class ActivityMappingService(
     }
   }
 
+  suspend fun updateScheduleMappings(updateRequest: ActivityMappingDto): ActivityMappingDto {
+    val activityScheduleId = if (activityMappingRepository.existsById(updateRequest.activityScheduleId)) {
+      updateRequest.activityScheduleId
+    } else {
+      throw NotFoundException("Activity schedule id=${updateRequest.activityScheduleId}")
+    }
+
+    val existingMappings = activityScheduleMappingRepository.findAllByActivityScheduleId(activityScheduleId)
+
+    // handle updates and deletes
+    existingMappings.forEach { existingMapping ->
+      updateRequest.scheduledInstanceMappings.findRequestedMapping(existingMapping)
+        ?.also { requestedMapping -> existingMapping.saveIfChanged(requestedMapping) }
+        ?: also { activityScheduleMappingRepository.delete(existingMapping) }
```

Comment:
why do we have to do all this manually - shouldn't hibernate do it for us by just passing in the new mappings and it then works out what is new / changed / deleted?

Reply:
Because we're using Spring Data JDBC which deletes all of the children and recreates them (unlike Spring Data JPA). I don't want that behaviour hence doing it manually.